### PR TITLE
PSA - minor fixes in all.sh

### PR DESCRIPTION
**Pull request by AndrzejKurek**
This PR handles changes requested by @gilles-peskine-arm here: https://github.com/ARMmbed/mbedtls/pull/2395#pullrequestreview-199710097

### CODE REVIEW

File: tests/scripts/all.sh [hunk 1]
```diff
@@ -820,25 +822,25 @@ component_test_submodule_make () {
...
     # Don't USE_CRYPTO_SUBMODULE: check that the submodule is not used with make
-    msg "build: make, full config - USE_CRYPTO_SUBMODULE, gcc+debug"
+    msg "build: make, full config without USE_CRYPTO_SUBMODULE, gcc+debug"
     scripts/config.pl full
     make CC=gcc CFLAGS='-g'
     msg "test: submodule libmbedcrypto wasn't built (USE_CRYPTO_SUBMODULE, make)"
     if_build_succeeded not test -f crypto/library/libmbedcrypto.a
     msg "test: libmbedcrypto symbols are from library files (USE_CRYPTO_SUBMODULE, make)"
-    if_build_succeeded objdump -g library/libmbedcrypto.a | grep -E 'library$' | not grep 'crypto' > /dev/null
+    if_build_succeeded objdump -g library/libmbedcrypto.a | if_build_succeeded grep -E 'library$' | if_build_succeeded not grep 'crypto' > /dev/null
```
Comment (gilles-peskine-arm) [inline tests/scripts/all.sh]:
```suggestion
    if_build_succeeded objdump_must_contain library/libmbedcrypto.a 'library$' crypto
```
and likewise below

with
```sh
# objdump_must_contain OBJECT_FILE MANDATORY_RE [MATCHING_LINES_MUST_NOT_INCLUDE]
objdump_must_contain () {
    objdump -g "$1" | grep -E "$2" |
    if [ -n "$3" ]; then ! grep -q -E "$3"; else cat; fi
}
```

Comment (AndrzejKurek) [reply tests/scripts/all.sh]: Done, thanks!

**Comment by mpg** [comment]
@AndrzejKurek This conflicts with #2403 which has higher priority. So we should wait until #2403 is merged, then rebased this one on development, before we review it again and get it merged.

**Comment by mpg** [comment]
@AndrzejKurek @gilles-peskine-arm  I went over these changes and found that all of them were related to having crypto as a submodule. Since that's no longer the case, I think this PR is no longer useful, so I'm closing it.

**Pull request closed without merging**